relays/headers-relay/src/sync_loop.rs

```diff
@@ -387,9 +408,16 @@ pub fn run<P: HeadersSyncPipeline, TC: TargetClient<P>>(
 			// print progress
 			progress_context = print_sync_progress(progress_context, &sync);
 
+			// run maintain procedures
+			if maintain_required && source_client_is_online && target_client_is_online {
+				log::debug!(target: "bridge", "Maintaining headers sync loop");
+				maintain_required = false;
+				sync_maintain.maintain(&mut sync).await;
+			}
+
 			// If the target client is accepting requests we update the requests that
 			// we want it to run
-			if target_client_is_online {
+			if !maintain_required && target_client_is_online {
```

Comment:
What problems would we run into if this `!maintain_required` check weren't here?

Reply:
The general principle of headers sync is that there's at most 1 future that works with source/target client at any moment. So we can't run maintain if client is busy with other requests. If we'll keep source/target client running other requests, maintain may never happen.